### PR TITLE
Added a method to find the index for a surface with a given name

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Needed this to make it easier to update procedurally generated meshes with multiple surfaces.

### CODE REVIEW

File: doc/classes/ArrayMesh.xml [hunk 1]
```diff
@@ -40,18 +40,21 @@
 			<return type="void">
 			</return>
 			<description>
+				Centers the geometry
```
Comment (akien-mga) [inline doc/classes/ArrayMesh.xml]: Missing dot :)

File: doc/classes/ArrayMesh.xml [hunk 2]
```diff
@@ -77,12 +81,23 @@
 			<argument index="1" name="arg1" type="float">
 			</argument>
 			<description>
+				Will perform a UV unwrap on the [code]ArrayMesh[/code] to prepare the mesh for lightmapping/
```
Comment (akien-mga) [inline doc/classes/ArrayMesh.xml]: Typo, trailing `/` should be a dot.

**Comment by GeorgeS2019** [comment]
@BastiaanOlij 

can you have a look at this finding to see if this explain [why **ArrayMesh** is not able to set blendshape in **Csharp** => link to bug issue](https://github.com/godotengine/godot/issues/36243) 

=================================================

In Csharp,  BlendShape are get and set through two classes
ArrayMesh.cs and VisualServer.cs

Interestingly, in ArrayMesh.cs there is no corresponding SetBlendShapeWeight method
Could this explain why the above bug happens?


================================================
There is this method in VisualServer.cs

```csharp
/// <summary>
/// <para>Sets the weight for a given blend shape associated with this instance.</para>
/// </summary>
[GodotMethod("instance_set_blend_shape_weight")]
public static void InstanceSetBlendShapeWeight(RID instance, int shape, float weight)
{
     NativeCalls.godot_icall_3_630(method_bind_262, ptr, RID.GetPtr(instance), shape, ref weight);
}
```

**Comment by BastiaanOlij** [comment]
Hmmm,

Doesn't look like ArrayMesh has any direct access to the blend shapes so I guess you can only set them when you add surfaces initially. Not sure how that is supposed to work or if we're just missing functions to set this.

I don't think the instance methods refer to meshes and can't take an ArrayMesh as a parameter. It's called from within MeshInstance so it will crash if you send a mesh to it.

**Pull request merged**